test/integration/Ipv8Connector.spec.ts

```diff
@@ -69,26 +71,25 @@ describe('Ipv8Connector.ts', function () {
     expect(did).to.eq(peers.employee.did)
   })
 
-  it('should be able to verify an attested claim', function (done) {
+  it('should be able to verify an attested claim', async function () {
     this.slow(5000)
     this.timeout(10000)
     const brewerConnector = new Ipv8Connector()
     const employeeConnector = new Ipv8Connector()
     brewerConnector.configure(peers.brewer.url)
     employeeConnector.configure(peers.employee.url)
 
-    employeeConnector.observeVerificationRequests(peers.employee.did, { did: peers.brewer.did }).then(result => {
-      const subscription = result.observable.subscribe({
-        next: c => {
-          employeeConnector.ipv8AttestationClient.allowVerify('eGU/YRXWJB18VQf8UbOoIhW9+xM=', c.claim.data)
-          subscription.unsubscribe()
-        },
-        error: e => assert.fail('Error when observing:' + e.message)
-      })
+    const observeResult = await employeeConnector.observeVerificationRequests(peers.employee.did, { did: peers.brewer.did })
+    await observeResult.readPromise
+    const subscription = observeResult.observable.subscribe({
+      next: c => {
+        employeeConnector.ipv8AttestationClient.allowVerify('eGU/YRXWJB18VQf8UbOoIhW9+xM=', c.claim.data)
+        subscription.unsubscribe()
```

Comment:
Wait, does this actually work? Is the subscription in scope?

Reply:
Yes it is :) `const` is block-scoped and lambda's don't have there own `this`, they are  effectively bound to the scope of the enclosing function. If the `subscription` was not in scope the test would fail with a `ReferenceError`.